src/error.rs

```diff
@@ -36,18 +38,31 @@ impl Error {
         Error(ErrorImpl { code, offset })
     }
 
+    #[cfg(feature = "std")]
     pub(crate) fn io(error: io::Error) -> Error {
         Error(ErrorImpl {
             code: ErrorCode::Io(error),
             offset: 0,
         })
     }
 
+    pub(crate) fn scratch_too_small() -> Error {
+        Error(ErrorImpl {
+            code: ErrorCode::ScratchTooSmall,
+            offset: 0,
```

Comment:
Can't we give an offset for this?

Reply:
We probably could! I will look into this